central/compliance/datastore/internal/store/postgres/store.go

```diff
@@ -0,0 +1,249 @@
+	cacheLock.Lock(domainID)
+	defer cacheLock.Unlock(domainID)
+
+	cachedDomain := domainCache.Get(domainID)
+	if cachedDomain != nil {
+		return cachedDomain.(*storage.ComplianceDomain), nil
+	}
+
+	domain, exists, err := s.domain.Get(ctx, domainID)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, errox.NotFound.Newf("domain with id %q was not found", domainID)
+	}
+	domainCache.Add(domainID, domain)
+	return domain, nil
+}
+
+func (s *storeImpl) getResultsFromMetadata(ctx context.Context, metadata *storage.ComplianceRunMetadata, flags types.GetFlags) (*storage.ComplianceRunResults, error) {
+	domain, err := s.getDomain(ctx, metadata.GetDomainId())
+	if err != nil {
+		return nil, err
+	}
+	results, exists, err := s.results.Get(ctx, metadata.GetRunId())
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, errox.NotFound.Newf("run results with id %q was not found", metadata.GetRunId())
+	}
+
+	results.RunMetadata = metadata
+	results.Domain = domain
+
+	if flags&(dsTypes.WithMessageStrings|dsTypes.RequireMessageStrings) != 0 {
+		externalizedStrings, exists, err := s.strings.Get(ctx, metadata.GetRunId())
+		if err != nil || !exists {
+			if flags&dsTypes.RequireMessageStrings != 0 {
+				return nil, errors.Wrap(err, "loading message strings")
+			}
+			log.Errorf("Could not load message strings for compliance run results: %v", err)
+		}
+		if !store.ReconstituteStrings(results, externalizedStrings) {
+			return nil, errors.New("some message strings could not be loaded")
+		}
+	}
+	return results, nil
+}
+
+func (s *storeImpl) GetSpecificRunResults(ctx context.Context, _, _, runID string, flags types.GetFlags) (types.ResultsWithStatus, error) {
+	metadata, exists, err := s.metadata.Get(ctx, runID)
+	if err != nil {
+		return types.ResultsWithStatus{}, err
+	}
+	if !exists {
+		return types.ResultsWithStatus{}, errox.NotFound.Newf("run metadata with id %q was not found", runID)
+	}
+	if !metadata.GetSuccess() {
+		return types.ResultsWithStatus{
+			FailedRuns: []*storage.ComplianceRunMetadata{
+				metadata,
+			},
+		}, nil
+	}
+	results, err := s.getResultsFromMetadata(ctx, metadata, flags)
+	if err != nil {
+		return types.ResultsWithStatus{}, err
+	}
+	return types.ResultsWithStatus{
+		LastSuccessfulResults: results,
+	}, nil
+}
+
+func (s *storeImpl) GetLatestRunResults(ctx context.Context, clusterID, standardID string, flags types.GetFlags) (types.ResultsWithStatus, error) {
+	metadataBatch, err := s.getLatestRunMetadata(ctx, clusterID, standardID)
+	if err != nil {
+		return types.ResultsWithStatus{}, nil
+	}
+	if metadataBatch.LastSuccessfulRunMetadata == nil {
+		return types.ResultsWithStatus{
+			FailedRuns: metadataBatch.FailedRunsMetadata,
+		}, nil
+	}
+	results, err := s.getResultsFromMetadata(ctx, metadataBatch.LastSuccessfulRunMetadata, flags)
+	if err != nil {
+		return types.ResultsWithStatus{}, nil
```

Comment:
Shouldn't we return err here?  Later when this thing is called it checks if it returns err.

Reply:
yikes, thanks